### PR TITLE
Fixed async of ChainModifier.modify.

### PR DESCRIPTION
**Pull request by bre1470**
Fixed error handling and async processing, which gets triggered by `DataTable.setModifier`.

### CODE REVIEW

**Review by karolkolodziej — APPROVED** [review]

**Review by pawellysy — APPROVED** [review]
LGTM!

File: ts/Data/Modifiers/ChainModifier.ts [hunk 1]
```diff
@@ -151,6 +151,56 @@ class ChainModifier extends DataModifier {
...
+                modifier.modify(chainTable.modified as T, eventDetail)
+            );
+        }
+
+        promiseChain = promiseChain.then((chainTable): T => {
+            table.modified = chainTable.modified;
+            return table;
+        });
+
+        promiseChain = promiseChain['catch']((error): Promise<T> => {
```
Comment (goransle) [inline ts/Data/Modifiers/ChainModifier.ts:192]: I might be missing something, but could this not be a `.catch` on the assignment above?

**Review by sebastianbochan — APPROVED** [review]

File: ts/Data/Modifiers/ChainModifier.ts (hunk 1 above)
Comment (bre1470) [reply ts/Data/Modifiers/ChainModifier.ts:192]: Kind of, but I have to use brackets for IE and then I can not make proper alignments and it becomes ugly code. So this is an aesthetical thing to make the code more readable.

Comment (goransle) [reply ts/Data/Modifiers/ChainModifier.ts:192]: Ah, IE 😢

**Review by goransle — APPROVED** [review]

**Pull request merged**